tools/insert.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/sh
+# Shell script equivalent of Metamath INSERT tool
+# Mario Carneiro 5-Jan-2022
+
+usage() {
+  cat << HELP
+INSERT - Insert a string at a specified column in each line of a file
+
+This command inserts a string at a specified column in each line
+in a file.  It is intended to aid further processing of column-
+sensitive files.  Note: the index of the first column is 1, not 0.  If a
+line is shorter than COLUMN, then it is padded with spaces so that
+STRING is still added at COLUMN.
+Syntax:  insert.sh STRING COLUMN < INFILE > OUTFILE
+HELP
+}
+
+if [ $# -eq 1 ] && [ "$1" = "-h" ]; then usage; exit; fi
+isNum() { case "$1" in '' | '0' | *[!0-9]*) return 1;; esac; }
```

Comment:
00 isAllowed? verify

Reply:
it is, the error checking isn't air tight here but I don't think it's necessary to be. If you pass 0 then it treats it as 1. (Probably passing huge numbers will fail too.) I'm not sure what kind of environment (i.e. what bash-isms) I'm allowed to assume of the user here. If you have a better suggestion feel free to improve it.